object-utils.go

```diff
@@ -73,27 +73,32 @@ func IsValidBucketName(bucket string) bool {
 // - Caret ("^")
 // - Grave accent / back tick ("`")
 // - Vertical bar / pipe ("|")
+//
```

Comment:
Keep isValidObjectName() more stricter in terms of object name verification and make isValidObjectPrefix() as the simpler one where it supports empty characters as well. 

Right now both functions have scattered code and it's not necessary to move it like that.